api_app/connectors_manager/classes.py

```diff
@@ -51,3 +54,41 @@ def after_run(self):
 
     def __repr__(self):
         return f"({self.connector_name}, job: #{self.job_id})"
+
+    @classmethod
+    def get_healthcheck_url_loc(cls) -> Dict[str, str]:
+        """
+        Should return url location: in config/secrets or given directly (url)
+          "secrets/config/url": "value"
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def health_check(cls, cc: ConnectorConfig) -> Optional[bool]:
+        """
+        basic health check: if instance is up or not (timeout - 10s)
+        raises: NotImplementedError
+          if get_healthcheck_url_loc is not overridden by subclass
+        """
+
+        url_loc = cls.get_healthcheck_url_loc()
+        url, health_status = None, None
+
+        if url_loc.get("url", None) is not None:
+            url = url_loc["url"]
+        elif url_loc.get("secrets", None) is not None:
+            secret_dict = cc._read_secrets(url_loc["secrets"])
+            url = secret_dict[url_loc["secrets"]]
+        elif url_loc.get("config", None) is not None:
+            url = cc.config[url_loc["config"]]
```

Comment:
I do not really get why this is implemented in such a complex manner.

Reply:
As I mentioned yesterday, I think passing the `cc: ConnectorConfig` argument is redundant. We can just hardcode the connector name inside the classes and use `ConnectorConfig.get(name)` and then `read_secrets["url_key_name"]`

Reply:
We can just hardcode: `url = cc._read_secrets["url_key_name"]`